frontend/www/js/omegaup/graderv2/GraderStore.ts

```diff
@@ -137,5 +138,507 @@ const storeOptions: StoreOptions<GraderStore> = {
+      } = value;
+
+      store.commit('request.input.interactive.idl', idl);
+      store.commit('InteractiveLanguage', language);
+      store.commit('InteractiveModuleName', module_name);
+      store.commit('request.input.interactive.main_source', main_source);
+      // if its the same template from before, no need to update
+      if (templates == state.request.input.interactive?.templates) {
+        state.dirty = true;
+        return;
+      }
+
+      // update with interactive problem templates for each language
+      // dont forget to update all cppxx and pyx templates
+      for (const extension in templates) {
+        if (Object.prototype.hasOwnProperty.call(templates, extension)) {
+          for (const language of Util.extensionToLanguages[extension]) {
+            interactiveTemplates[language] = templates[extension];
+          }
+        } else {
+          for (const language of Util.extensionToLanguages[extension]) {
+            interactiveTemplates[language] =
+              originalInteractiveTemplates[extension];
+          }
+        }
+      }
+
+      state.dirty = true;
+    },
+    InteractiveLanguage(state: GraderStore, value: string) {
+      if (value == 'cpp') value = 'cpp17-gcc';
+      if (!state.request.input.interactive) return;
+      state.request.input.interactive.language = value;
+      state.dirty = true;
+    },
+    InteractiveModuleName(state: GraderStore, value: string) {
+      if (!state.request.input.interactive) return;
+      state.request.input.interactive.module_name = value;
+      state.dirty = true;
+    },
+    updatingSettings(state: GraderStore, value: boolean) {
+      state.updatingSettings = value;
+    },
+    createCase(
+      state: GraderStore,
+      caseData: { name: string; in: string; out: string; weight?: number },
+    ) {
+      // if case doesnt already exist create it?
+      // no! always create a case
+      // 2 cases can be of same name and different data
+
+      Vue.set(state.request.input.cases, caseData.name, {
+        in: caseData.in || '',
+        out: caseData.out || '',
+        weight: caseData.weight || 1,
+      });
+      // if we call this function, we must set current case
+      // or it could cause errors
+      state.currentCase = caseData.name;
+      state.dirty = true;
+    },
+    removeCase(state: GraderStore, name: string) {
+      if (
+        !Object.prototype.hasOwnProperty.call(state.request.input.cases, name)
+      )
+        return;
+
+      const keys = Object.keys(store.state.request.input.cases);
+
+      // do not delete if there is only one test case
+      if (keys.length === 1) {
+        return;
+      }
+
+      // switch to a random case
+      const caseName = keys[0] === name ? keys[1] : keys[0];
+      store.commit('currentCase', caseName);
+
+      Vue.delete(state.request.input.cases, name);
+      state.dirty = true;
+    },
+    limits(state: GraderStore, limits: types.LimitsSettings) {
```

Comment:
Nit:
```suggestion
    limits(_state: GraderStore, limits: types.LimitsSettings) {
```
If `state` is not being used you can use `_`